transcom/mymove
# MB-16790 Regenerate service items when address change updates pricing

## [Jira ticket](https://dp3.atlassian.net/browse/MB-16790)

## Summary
When an approved address change results in a conversion between shorthaul and linehaul, the system will reject previous service items and generate new ones. 

**Is there anything you would like reviewers to give additional scrutiny?**

- Let me know if there is a better way to clone a service item struct

### Setup to Run the Code

- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Access the office app as a prime user
2. Find a move and update the destination address to one that will change the move from linehaul to shorthaul and/or vice versa.
3. Approve the address change as a TOO
4. Confirm that the service items have been updated and that DLH was converted to DSH or vice versa
